server/static/job_list.jsx

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const e = React.createElement;
+
+class JobList extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { queued: [], running: [], completed: [] };
+    }
+
+    componentDidMount() {
```

Comment:
This looks very familiar with the above in hardware. Is it possible to make abstract base classes for these React elements and classes?